lib/src/operance_data_column.dart

```diff
@@ -50,8 +56,81 @@ class OperanceDataColumn<T> {
   /// Defaults to `false`.
   final bool sortable;
 
+  /// Indicates whether the column is filterable.
+  ///
+  /// Defaults to `false`.
+  final bool filterable;
+
   /// The width of the column.
   ///
   /// Defaults to an instance of [OperanceDataColumnWidth].
   final OperanceDataColumnWidth width;
+
+  /// A custom comparator function for sorting.
+  ///
+  /// If provided, this function will be used for sorting instead of the default
+  /// comparison. The function should return a negative value if a < b, zero if
+  /// a == b, and a positive value if a > b.
+  final int Function(T a, T b)? comparator;
+
+  /// A function to extract the raw value from an item for sorting and
+  /// filtering.
+  ///
+  /// This is more efficient than using [getSearchableValue] for large datasets.
```

Comment:
doc shouldn't include deprecated content, new devs who'll use this package won't know what getSearchableValue is. Also the example app included in the repo should be erroring out now since this has changed. ensure to adjust changes in the example app as well